tommath_private.h

```diff
@@ -31,7 +31,7 @@ extern "C" {
  */
 #ifdef MP_NO_ZERO_ON_FREE
 #  define MP_FREE_BUFFER(mem, size)   MP_FREE((mem), (size))
-#  define MP_FREE_DIGITS(mem, digits) MP_FREE((mem), sizeof (mp_digit) * (digits))
+#  define MP_FREE_DIGITS(mem, digits) MP_FREE((mem), sizeof (mp_digit) * (size_t)(digits))
 #else
 #  define MP_FREE_BUFFER(mem, size)   do { size_t fs_ = (size); void* fm_ = (mem); if (fm_) { MP_ZERO_BUFFER(fm_, fs_); MP_FREE(fm_, fs_); } } while (0)
 #  define MP_FREE_DIGITS(mem, digits) do { int fd_ = (digits); void* fm_ = (mem); if (fm_) { MP_ZERO_BUFFER(fm_, sizeof (mp_digit) * (size_t)fd_); MP_FREE(fm_, sizeof (mp_digit) * (size_t)fd_); } } while (0)
```

Comment:
Noticing that the multiplication "sizeof (mp_digit) * (size_t)fd_"is done twice in this macro. Better do the multiplication, store it in a variable, and them use that variable twice

Reply:
Ok. But probably this does not matter since compilers perform CSE and similar optimizations.

Reply:
This has also not much to do with this PR, since it was merged in #255. Maybe change it in #259? @fperrad is improving the macro there.

Reply:
Ok, well it has something to do since I am also adding the missing cast here ;)
Maybe @fperrad could also cherry-pick that commit from here... It is a bit of a mess if the same code is touched from multiple PRs.